Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_timeperiod.py

```diff
@@ -69,6 +69,11 @@ def token_before_date(self) -> str:
     def pure_number_regex(self) -> List[Pattern]:
         raise NotImplementedError
 
+    @property
+    @abstractmethod
+    def time_zone_extractor(self) -> DateTimeParser:
```

Comment:
Here the return type is not DateTimeParser or the property has the be renamed as _time_zone_parser_. Please, check it.

Reply:
True!, It was DateTimeExtractor, thanks.